File: app/src/main/java/com/example/smarthome/relays/models/Relay.java
package com.example.smarthome.relays.models;

import com.example.smarthome.WifiDevice;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;


public class Relay extends WifiDevice{

    @SerializedName("id")
    @Expose
    private Long id;

    public Relay(String name,  String ip, boolean on){
        super(name, ip, on);
    }


    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }
}

File: app/src/main/java/com/example/smarthome/relays/models/Relay.kt
package com.example.smarthome.relays.models

import com.example.smarthome.R
import com.example.smarthome.adapter.ViewType
import com.example.smarthome.WifiDevice


data class Relay(
        var id: Long?
) : WifiDevice(), ViewType {

    constructor(name: String, ip: String, on: Boolean)
            : this(null){
        super.setName(if (name != "") name else "No name")
        super.setIp(ip)
        super.setOn(on)
    }

    override fun getViewType(): Int {
       return R.layout.item_relay
    }
}